src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -1423,6 +1423,26 @@ static int patch_addr_in_movptr(address branch, address target) {
   return MOVPTR_INSTRUCTIONS_NUM * NativeInstruction::instruction_size;
 }
 
+static int patch_addr_in_movptr2(address instruction_address, address target) {
```

Comment:
Can we have a common entry of `patch_addr_in_movptr` which delegates work to `patch_addr_in_movptr1` and `patch_addr_in_movptr2`?

Reply:
I think it makes sense to split them up as the difference between movptr1 and movptr2 is already done in the caller then. And IIUC we don't plan to merge movptr1 and movptr2 together at any point, so we don't particularly need to abstract them away.

Reply:
All right. I can live with this.

Reply:
All right. I can live with this.